notify/impl.go

```diff
@@ -613,15 +619,24 @@ func (n *PagerDuty) notifyV2(
 
 	var buf bytes.Buffer
 	if err := json.NewEncoder(&buf).Encode(msg); err != nil {
-		return false, err
+		return false, fmt.Errorf("failed to encode PagerDuty v2 message: %v", err)
 	}
 
 	resp, err := post(ctx, c, n.conf.URL.String(), contentTypeJSON, &buf)
 	if err != nil {
-		return true, err
+		return true, fmt.Errorf("failed to post message to PagerDuty: %v", err)
 	}
 	defer resp.Body.Close()
 
+	// See: https://v2.developer.pagerduty.com/docs/events-api-v2#api-response-codes--retry-logic
+	if resp.StatusCode == http.StatusBadRequest {
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return true, fmt.Errorf("failed to parse response from PagerDuty: %v", err)
```

Comment:
The link you've provided says to not retry on 400BadRequest; shouldn't this be `false`?

Adding some more context to the error would also probably be helpful, like letting the user know that the failure was in parsing an error response.

Reply:
Oh sorry of course your right, I changed it to false. I also added some more context and replaced "failed to parse" with "failed to read" because the body is not actually parsed here.